cpp/arduino/EEPROM.h

```diff
@@ -0,0 +1,166 @@
+/*
+  EEPROM.h - EEPROM library
+  Original Copyright (c) 2006 David A. Mellis.  All right reserved.
+  New version by Christopher Andrews 2015.
+
+  Copy of https://github.com/arduino/ArduinoCore-megaavr/blob/c8a1dd996c783777ec46167cfd8ad3fd2e6df185/libraries/EEPROM/src/EEPROM.h
+  modified by James Foster in 2020 to work with Arduino CI.
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2.1 of the License, or (at your option) any later version.
```

Comment:
We should probably reconsider adding this directly since its license is inconsistent.

Reply:
I'm really out of my depth with regard to the terms of all the different licenses, and really can't offer advice one way or another.  I have no idea whether your own rewrite counts as some "derivative" in the way the licenses are written.

I think as long as we're doing our best and intend to respond to any legal complaints -- fixing the license as appropriate -- (which I do intend to do) then I'm not sure what else it would take to move forward.  In any case, if we keep the file let's keep the license and it can all get sorted out later.

Reply:
As owner of the project, you certainly have the last word on this. But (did you sense a "but" coming?), the difference between the GNU license and your Apache license is night and day (with _slight_ exaggeration, think Catholics and Protestants in 16th, 17th, and 18th century Europe, or Antifa vs. Proud Boys today). While I can't promise that my replacement would withstand a legal challenge, I'm pretty confident that it would be better to leave out the GNU reference. Other than the file name and the published API, I made an effort to write the code from scratch and think that the risk is extremely small. By leaving in the GNU license, you risk attaching GNU to my code, and then you would need another rewrite.

Reply:
I saw your rewrite after adding my comment, and I think your rewrite satisfies